OneSignalSDK/onesignal/in-app-messages/src/main/java/com/onesignal/inAppMessages/internal/backend/impl/InAppBackendService.kt

```diff
@@ -209,4 +201,59 @@ internal class InAppBackendService(
     ) {
         Logging.error("Encountered a $statusCode error while attempting in-app message $requestType request: $response")
     }
+
+    private suspend fun attemptFetchWithRetries(baseUrl: String, offset: Long): List<InAppMessage>? {
+        var attempts = 1
+
+        while (true) {
+            val urlWithOffset = "$baseUrl?offset=$offset"
+            val response = _httpClient.get(urlWithOffset)
+
+            if (response.isSuccess) {
+                val jsonResponse = response.payload?.let { JSONObject(it) }
+                return jsonResponse?.let { hydrateInAppMessages(it) }
+            } else if (response.statusCode == 429) { // 429 Too Many Requests
```

Comment:
I believe the spec defined 425 "Too Early". However we should handle both 425 and 429